clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/controllers/GoogleLoadBalancerController2.groovy

```diff
@@ -0,0 +1,145 @@
+      def summary = new GoogleLoadBalancerAccountRegionSummary(name: name)
+
+      views.each { GoogleLoadBalancer2.View view ->
+        summary.mappedAccounts[view.account].mappedRegions[view.region].loadBalancers << new GoogleLoadBalancerSummary(
+            account: view.account,
+            region: view.region,
+            name: view.name)
+      }
+
+      summary
+    }
+  }
+
+  @RequestMapping(value = "/{name:.+}", method = RequestMethod.GET)
+  GoogleLoadBalancerAccountRegionSummary get(@PathVariable String name) {
+    // TODO(ttomsu): It's inefficient to pull everything back and (possibly) discard most of it.
+    // Refactor when addressing https://github.com/spinnaker/spinnaker/issues/807
+    list().find { it.name == name }
+  }
+
+  @RequestMapping(value = "/{account}/{region}/{name:.+}", method = RequestMethod.GET)
+  List<GoogleLoadBalancerDetails> getDetailsInAccountAndRegionByName(@PathVariable String account,
+                                                                     @PathVariable String region,
+                                                                     @PathVariable String name) {
+    GoogleLoadBalancer2.View view = googleLoadBalancerProvider.getApplicationLoadBalancers(name).find { view ->
+      view.account == account && view.region == region
+    }
+
+    [new GoogleLoadBalancerDetails(loadBalancerName: view.name,
+                                   createdTime: view.createdTime,
+                                   dnsname: view.ipAddress,
+                                   ipAddress: view.ipAddress,
+                                   listenerDescriptions: [[
+                                       listener: new ListenerDescription(instancePort: view.portRange,
+                                                                         loadBalancerPort: view.portRange,
+                                                                         instanceProtocol: view.ipProtocol,
+                                                                         protocol: view.ipProtocol)
+                                   ]])]
+  }
+
+  static class GoogleLoadBalancerAccountRegionSummary {
+
+    String name
+
+    @JsonIgnore
+    Map<String, GoogleLoadBalancerAccount> mappedAccounts = [:].withDefault {
+      String accountName -> new GoogleLoadBalancerAccount(name: accountName)
+    }
+
+    @JsonProperty("accounts")
+    List<GoogleLoadBalancerAccount> getAccounts() {
+      mappedAccounts.values() as List
+    }
+  }
+
+  static class GoogleLoadBalancerAccount {
+
+    String name
+
+    @JsonIgnore
+    Map<String, GoogleLoadBalancerAccountRegion> mappedRegions = [:].withDefault {
+      String region -> new GoogleLoadBalancerAccountRegion(name: region)
+    }
+
+    @JsonProperty("regions")
+    List<GoogleLoadBalancerAccountRegion> getRegions() {
+      mappedRegions.values() as List
+    }
+  }
+
+  static class GoogleLoadBalancerAccountRegion {
+    String name
```

Comment:
eh, I guess I can live with it. Looked weird at first read, but I suppose it identifying the _name_ of the region is ok.